api/v1alpha1/proxydefaults_types_test.go

```diff
@@ -0,0 +1,625 @@
+		"envoy_tracing_json, protocol & members": json.RawMessage(`{"envoy_tracing_json": "{\"http\":{\"name\":\"envoy.zipkin\",\"config\":{\"collector_cluster\":\"zipkin\",\"collector_endpoint\":\"/api/v1/spans\",\"shared_span_context\":false}}}","protocol":  "http", "members": 3}`),
+	}
+	for name, c := range cases {
+		proxyDefaults := ProxyDefaults{
+			ObjectMeta: metav1.ObjectMeta{
+				Name: "global",
+			},
+			Spec: ProxyDefaultsSpec{
+				Config: c,
+			},
+		}
+		t.Run(name, func(t *testing.T) {
+			require.Nil(t, proxyDefaults.validateConfig(nil))
+		})
+	}
+}
+
+func TestProxyDefaults_ValidateConfigInvalid(t *testing.T) {
+	cases := map[string]json.RawMessage{
+		"non_map json": json.RawMessage(`"{\"http\":{\"name\":\"envoy.zipkin\",\"config\":{\"collector_cluster\":\"zipkin\",\"collector_endpoint\":\"/api/v1/spans\",\"shared_span_context\":false}}}"`),
+		"yaml":         json.RawMessage(`protocol: http`),
+		"json array":   json.RawMessage(`[1,2,3,4]`),
+		"json literal": json.RawMessage(`1`),
+	}
+	for name, c := range cases {
+		proxyDefaults := ProxyDefaults{
+			ObjectMeta: metav1.ObjectMeta{
+				Name: "global",
+			},
+			Spec: ProxyDefaultsSpec{
+				Config: c,
+			},
+		}
+		t.Run(name, func(t *testing.T) {
+			require.Equal(t, proxyDefaults.validateConfig(field.NewPath("spec")).Detail, "must be valid map value")
+		})
+	}
+}
+
+func TestProxyDefaults_AddFinalizer(t *testing.T) {
+	resolver := &ProxyDefaults{}
+	resolver.AddFinalizer("finalizer")
+	require.Equal(t, []string{"finalizer"}, resolver.ObjectMeta.Finalizers)
+}
+
+func TestProxyDefaults_RemoveFinalizer(t *testing.T) {
+	resolver := &ProxyDefaults{
+		ObjectMeta: metav1.ObjectMeta{
+			Finalizers: []string{"f1", "f2"},
+		},
+	}
+	resolver.RemoveFinalizer("f1")
+	require.Equal(t, []string{"f2"}, resolver.ObjectMeta.Finalizers)
+}
+
+func TestProxyDefaults_SetSyncedCondition(t *testing.T) {
+	resolver := &ProxyDefaults{}
+	resolver.SetSyncedCondition(corev1.ConditionTrue, "reason", "message")
+
+	require.Equal(t, corev1.ConditionTrue, resolver.Status.Conditions[0].Status)
+	require.Equal(t, "reason", resolver.Status.Conditions[0].Reason)
+	require.Equal(t, "message", resolver.Status.Conditions[0].Message)
+	now := metav1.Now()
+	require.True(t, resolver.Status.Conditions[0].LastTransitionTime.Before(&now))
+}
+
+func TestProxyDefaults_GetSyncedConditionStatus(t *testing.T) {
+	cases := []corev1.ConditionStatus{
+		corev1.ConditionUnknown,
+		corev1.ConditionFalse,
+		corev1.ConditionTrue,
+	}
+	for _, status := range cases {
+		t.Run(string(status), func(t *testing.T) {
+			resolver := &ProxyDefaults{
+				Status: Status{
+					Conditions: []Condition{{
+						Type:   ConditionSynced,
+						Status: status,
+					}},
+				},
+			}
+
+			require.Equal(t, status, resolver.SyncedConditionStatus())
+		})
+	}
+}
+
+func TestProxyDefaults_GetConditionNil(t *testing.T) {
+	resolver := &ProxyDefaults{}
+	require.Nil(t, resolver.GetCondition(ConditionSynced))
+}
```

Comment:
Can you also grab the missing tests from https://github.com/hashicorp/consul-k8s/pull/332/files#diff-6ec9b8708393dcab4ff2c09614afe4ae